hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -479,6 +491,21 @@ boolean shouldStartJobs() {
                 && !partitionService.hasOnGoingMigrationLocal();
     }
 
+    private boolean checkAllMembersHaveTheSameState(ClusterState clusterState) {
+        Set<Member> members = nodeEngine.getClusterService().getMembers();
+        InternalOperationService operationService = nodeEngine.getOperationService();
+        List<Future<ClusterMetadata>> futures =
+                members.stream()
+                       .filter(m -> !m.localMember())
+                       .map(m -> {
+                           Future<ClusterMetadata> future = operationService.invokeOnTarget(JetService.SERVICE_NAME,
+                                   new GetClusterMetadataOperation(), m.getAddress());
+                           return future;
+                       })
+                       .collect(toList());
+        return futures.stream().map(f -> uncheckCall(f::get)).allMatch(metaData -> metaData.getState() == clusterState);
```

Comment:
I think `scheduleWithRepetition` cancels the repetition if some execution fails. This is likely to fail on cluster changes, we should catch the exceptions.